salt/crypt.py

```diff
@@ -377,6 +378,19 @@ def __singleton_init__(self, opts, io_loop=None):
         else:
             self.authenticate()
 
+    def __deepcopy__(self, memo):
+        cls = self.__class__
+        result = cls.__new__(cls, copy.deepcopy(self.opts, memo), io_loop=None)
+        memo[id(self)] = result
+        for key, value in self.__dict__.items():
```

Comment:
If you are going to do deepcopy of the value, I'm not certain you need .items() (as that will make a shallow copy as well).